spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -237,6 +237,13 @@ public boolean dropTable(Identifier ident) {
     return icebergCatalog.dropTable(ident) || getSessionCatalog().dropTable(ident);
   }
 
+  @Override
+  public boolean purgeTable(Identifier ident) {
+    // no need to check table existence to determine which catalog to use. if a table doesn't exist then both are
+    // required to return false.
+    return icebergCatalog.purgeTable(ident) || getSessionCatalog().purgeTable(ident);
```

Comment:
Quick check here, we are safe in the GC_ENABLED situation because iceberg.purge will throw a runtime exception correct? Just want to make sure we don't call purge in that case.

Reply:
That's right. The `icebergCatalog.purgeTable` will throw a RuntimeException if it is an iceberg table. Otherwise, it depends on the `sessionCatalog` implementation because is out of the scope of iceberg.